tests/test_csm.py

```diff
@@ -0,0 +1,69 @@
+from unittest import mock
+import pytest
+
+from plio.io.io_gdal import GeoDataset
+
+import csmapi
+from knoten import csm
+
+@pytest.fixture
+def mock_sensor():
+    mock_sensor = mock.MagicMock(spec=csmapi.RasterGM)
+    return mock_sensor
+
+@pytest.fixture
+def pt():
+    return csmapi.ImageCoord(0.0, 0.0)
+
+def test_generate_ground_point_with_float(mock_sensor):
+    csm.generate_ground_point(0, (0.5, 0.5), mock_sensor)
+    # The internal conversion from tuple to csmapi.ImageCoord means 
+    # assert_called_once_with fails due to different addresses of
+    # different objects.
+    mock_sensor.imageToGround.assert_called_once()
+
+def test_generate_ground_point_with_imagecoord(mock_sensor, pt):
+    height = 0.0
+    csm.generate_ground_point(height, pt, mock_sensor)
+    mock_sensor.imageToGround.assert_called_once_with(pt, height)
+
+@mock.patch.object(csm, 'get_radii', return_value=(10,10))
+@mock.patch('pyproj.transformer.Transformer.transform', return_value=(0,0,0))
+@mock.patch.object(csm, '_compute_intersection_distance', return_value=0)
+def test_generate_ground_point_with_dtm(mock_sensor, pt, _):
```

Comment:
As far as I can tell, pytest is counting the number of decorators and then assuming that n-decorators == n-args to the func. To get the test parser to work, I added the `_` arg. This seems super janky. Any thoughts on whether this approach should be different?

Reply:
I'm not sure here, you might try putting the other `mock.patch.object` above the `mock.patch`. Some of pythons test suite is very order dependent

Reply:
This comment was 💰. The mock.patch.object needs to be passed into the test as an arg. The order is also important where the mocked stuff needs to be first in the test signature, then any fixtures. 

Local results with this setup: `5 passed, 4 warnings in 1.89s`